src/index.js

```diff
@@ -86,9 +86,7 @@ export function asObject(shape) {
       }
 
       const out = {}
-      const keys = Object.keys(raw)
-      for (let i = 0; i < keys.length; ++i) {
-        const key = keys[i]
+      for (const key of Object.keys(raw)) {
```

Comment:
Doesn't this change take some performance hit?

Reply:
Because we have enabled the `['@babel/plugin-transform-for-of', { assumeArray: true }]` setting, this translates to roughly the same code as before, with `keys` being assigned up-front and a loop counter going from `0` to `length`, but now Babel does it instead of us.

Reply:
I put it back the way it was.